### ISSUE TITLE
[BUG] Adherence to google java style guide

### ISSUE THREAD

**Issue by bodograumann** [OpenAPITools/openapi-generator#4322]
Maybe I'm understanding this wrong, but according to the [Guidelines for Contributing](https://github.com/OpenAPITools/openapi-generator/blob/master/CONTRIBUTING.md), java code should adhere to the [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html).

The most glaring difference is that the style guide defines an indentation of *2 spaces*, while most java-sources in this repo use *4 spaces*.

Personally I would prefer a tool doing as many formatting and checking as possible. E.g.
```bash
prettier SomeClassName.java --print-width=100 --write
```

Most importantly it is unclear to me, how new code should be formatted.

**Comment by wing328** [comment]
> prettier SomeClassName.java --print-width=100 --write

You can do that by enabling file post-processing. When you run the Java client generator, there are tips in the output showing you how to do it and you can use your favorite code formatter.

**Comment by bodograumann** [comment]
I am not talking about java code generated by one of the java templates. This issue concerns the code of the openapi generator itself.

**Comment by wing328** [comment]
I agree all the Java code in the project itself should conform to a certain style guide.

I usually use Intellji IDEA to format the code. Maybe we should use that as a reference and list Intellji IDEA as the recommended IDE.

We tried to enforce style guide using a Java plugin but lots of PRs failed the check and people simply don't have time to update their PR so that's why we didn't enforce it in projects at the moment.

**Comment by ahl** [comment]
What would you think about deleting the two unused style formats: `eclipse-formatter.xml` and `google_checkstyle.xml`

I found it confusing that neither of these was in use so wasted a bit of time trying to use them.
